Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2AckermannModelHook.cpp

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <PhysX/EditorColliderComponentRequestBus.h>
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <ROS2/RobotControl/ControlConfiguration.h>
+#include <RobotControl/Controllers/AckermannController/AckermannControlComponent.h>
+#include <RobotControl/ROS2RobotControlComponent.h>
+#include <RobotImporter/SDFormat/ROS2ModelPluginHooks.h>
+#include <RobotImporter/SDFormat/ROS2SDFormatHooksUtils.h>
+#include <Source/EditorColliderComponent.h>
+#include <Source/EditorHingeJointComponent.h>
+#include <VehicleDynamics/ModelComponents/AckermannModelComponent.h>
+#include <VehicleDynamics/Utilities.h>
+#include <VehicleDynamics/WheelControllerComponent.h>
+
+#include <gz/math/Vector2.hh>
+#include <gz/math/Vector3.hh>
+#include <sdf/Joint.hh>
+#include <sdf/Link.hh>
+#include <sdf/Model.hh>
+#include <sdf/Plugin.hh>
+
+namespace ROS2::SDFormat
+{
+    namespace AckermannParser
+    {
+        static constexpr float epsilon = 0.001f;
+
+        namespace Wheels
+        {
+            static constexpr unsigned int FrontLeft = 0;
+            static constexpr unsigned int FrontRight = 1;
+            static constexpr unsigned int RearLeft = 2;
+            static constexpr unsigned int RearRight = 3;
+            static constexpr unsigned int SteeringLeft = 4;
+            static constexpr unsigned int SteeringRight = 5;
+            static constexpr unsigned int Total = 6;
+        } // namespace Wheels
+
+        AZ::Outcome<AZ::Vector3, AZStd::string> GetPosition(const AZ::Entity* entity)
+        {
+            if (entity)
+            {
+                auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>();
+                if (transformInterface)
```

Comment:
Could be combined into one line 
```suggestion
                if (auto* transformInterface = entity->FindComponent<AzToolsFramework::Components::TransformComponent>())
```